src/IotHub/IotHub/help/Set-AzIotHubMessageEnrichment.md

```diff
@@ -0,0 +1,230 @@
+﻿---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.IotHub.dll-Help.xml
+Module Name: Az.IotHub
+online version: https://docs.microsoft.com/en-us/powershell/module/az.iothub/set-aziothubmessageenrichment
+schema: 2.0.0
+---
+
+# Set-AzIotHubMessageEnrichment
+
+## SYNOPSIS
+Update a message enrichment in your IoT hub.
+
+## SYNTAX
+
+### ResourceSet (Default)
+```
+Set-AzIotHubMessageEnrichment [-ResourceGroupName] <String> [-Name] <String> [-Key] <String> [-Value <String>]
+ [-Endpoint <String[]>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### InputObjectSet
+```
+Set-AzIotHubMessageEnrichment [-InputObject] <PSIotHub> [-Key] <String> [-Value <String>]
+ [-Endpoint <String[]>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### ResourceIdSet
+```
+Set-AzIotHubMessageEnrichment [-ResourceId] <String> [-Key] <String> [-Value <String>] [-Endpoint <String[]>]
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+For a detailed explanation of message enrichments in Azure IoT Hub, see https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-message-enrichments-overview
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Set-AzIotHubMessageEnrichment -ResourceGroupName "myresourcegroup" -Name "myiothub" -Key "newKey" -Value "updatedValue"
+
+Key         : newKey
```

Comment:
These should show the table formats

Reply:
The output is in table-format if there is more than one record otherwise it will show as list-format. As Set cmdlet is applicable for one record only so the output shows in list-format.

Reply:
See above.  Not the recommended pattern